### PR TITLE
editoast: add endpoint to retrieve electrical profiles along a path

### PR DESCRIPTION
**Pull request by Castavo**
This PR can be reviewed commit by commit.

The first commit refactors the `catenaries_on_path` endpoint so that the functions can be reused for the new and very similar endpoint. 
The second adds said endpoint.

### CODE REVIEW

File: editoast/src/views/pathfinding/electrical_profiles.rs [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+pub fn routes() -> impl HttpServiceFactory {
+    services![electrical_profiles_on_path,]
+}
+
+/// Build a rangemap for each track section, giving the electrical profile for each range
+fn map_electrical_profiles(
+    electrical_profile_set: ElectricalProfileSetData,
+    power_class: String,
+    track_ids: HashSet<String>,
+) -> (HashMap<String, RangeMap<Float, String>>, Vec<InternalError>) {
```
Comment (flomonster) [inline editoast/src/views/pathfinding/electrical_profiles.rs]: We could make a type of it.

File: editoast/src/views/pathfinding/electrical_profiles.rs [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+    web::{Data, Json, Path, Query},
+};
+use rangemap::RangeMap;
+use serde::{Deserialize, Serialize};
+use std::collections::{HashMap, HashSet};
+use std::fmt::Debug;
+
+pub fn routes() -> impl HttpServiceFactory {
+    services![electrical_profiles_on_path,]
+}
```
Comment (flomonster) [inline editoast/src/views/pathfinding/electrical_profiles.rs]: Did you try to use route_v2 (to generate the openapi using utoipa ?).
If that's too much trouble, we can leave it at that for now.

File: editoast/src/views/pathfinding/electrical_profiles.rs [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+
+    #[fixture]
+    async fn electrical_profile_set(db_pool: Data<DbPool>) -> TestFixture<ElectricalProfileSet> {
+        let ep_data = ElectricalProfileSetData {
+            levels: vec![
+                ElectricalProfile {
+                    value: "A".into(),
+                    power_class: "1".into(),
+                    track_ranges: vec![
+                        TrackRange {
```
Comment (flomonster) [inline editoast/src/views/pathfinding/electrical_profiles.rs]: We could add a constructor like `TrackRange::new<T: AsRef<str>>(...)` to simplify this part.

File: editoast/src/views/pathfinding/rangemap_utils.rs
Comment (flomonster) [inline editoast/src/views/pathfinding/rangemap_utils.rs:1]: We could make a library of it. @leovalais What do you think?

**Review by flomonster — APPROVED** [review]

File: editoast/src/views/pathfinding/rangemap_utils.rs
Comment (leovalais) [reply editoast/src/views/pathfinding/rangemap_utils.rs:1]: Definitely

File: editoast/src/views/pathfinding/electrical_profiles.rs (hunk 2 above)
Comment (leovalais) [reply editoast/src/views/pathfinding/electrical_profiles.rs]: +1, you can use #5355 as an example, call me if you have any trouble using routes! :)

File: editoast/src/views/pathfinding/electrical_profiles.rs [hunk 4]
```diff
@@ -0,0 +1,324 @@
...
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct ProfilesOnPathResponse {
+    electrical_profile_ranges: Vec<RangedValue>,
+    warnings: Vec<InternalError>,
+}
+
+#[get("/{path_id}/electrical_profiles")]
+async fn electrical_profiles_on_path(
```
Comment (leovalais) [inline editoast/src/views/pathfinding/electrical_profiles.rs:92]: Could you add an `utoipa::path` annotation on this endpoint. Even if you don't use routes! and the annotation is left unused, that'll make the migration to utoipa easier

**Review by leovalais — APPROVED** [review]

**Review by clarani — APPROVED** [review]
LGTM, tested :heavy_check_mark:

**Review by leovalais — APPROVED** [review]
lgtm

**Pull request merged**